packages/timeline-state-resolver/src/conductor.ts

```diff
@@ -621,33 +649,45 @@ export class Conductor extends EventEmitter<ConductorEvents> {
 			if (this.devices.has(deviceId)) {
 				throw new Error(`Device "${deviceId}" already exists when creating device`)
 			}
+			throwIfAborted()
 			this.devices.set(deviceId, newDevice)
 
 			return newDevice
 		} catch (e) {
-			if (newDevice) {
-				try {
-					await newDevice.terminate()
-				} catch (e) {
-					this.emit('error', `Cleanup failed of aborted device "${newDevice.deviceId}": ${e}`)
-				}
-			}
+			await this.terminateUnwantedDevice(newDevice)
 			this.devices.delete(deviceId)
 			this.emit('error', 'conductor.createDevice', e)
 			return Promise.reject(e)
 		}
 	}
+
+	private async terminateUnwantedDevice(newDevice: DeviceContainer<DeviceOptionsBase<any>> | undefined) {
+		if (newDevice) {
+			try {
+				await newDevice.terminate()
+			} catch (e) {
+				this.emit('error', `Cleanup failed of aborted device "${newDevice.deviceId}": ${e}`)
+			}
+		}
```

Comment:
Just a suggestion. I understand if keeping the current code demonstrates the intent in a clearer way.
```suggestion
		await newDevice?.terminate()
			.catch(e => this.emit('error', `Cleanup failed of aborted device "${newDevice.deviceId}": ${e}`))
```